File: packages/enterprise/src/audit-console.ts
import type { AuditLogger, AuditEvent } from "./audit.js";

/**
 * Console audit logger (development); production should use structured logger or external sink.
 */
export class ConsoleAuditLogger implements AuditLogger {
  async log(event: AuditEvent): Promise<void> {
    const payload = JSON.stringify({
      ...event,
      timestamp: new Date(event.timestamp * 1000).toISOString(),
    });
    console.log(`[AUDIT] ${payload}`);
  }
}

File: python/src/agent_framework/enterprise/audit_console.py
"""Console audit logger."""

import json
from agent_framework.enterprise.audit import AuditLogger, AuditEvent


class ConsoleAuditLogger(AuditLogger):
    async def log(self, event: AuditEvent) -> None:
        payload = {
            "type": event.type,
            "timestamp": event.timestamp,
            "tenant_id": event.tenant_id,
            "user_id": event.user_id,
            "thread_id": event.thread_id,
            "metadata": event.metadata,
            "resource": event.resource,
        }
        print(f"[AUDIT] {json.dumps(payload)}")
